### PR TITLE
Speed up grid creation

### PR DESCRIPTION
**Pull request by mogres**
Problem
=======
Creating grids for mesh objects takes a long time due to checking for grid points inside/on the surface of the mesh. 

Solution
========
I vectorized the checking for grid points lying inside or on the surface of the mesh using numpy and trimesh.contains function. This can lead to a speed up of up to 4x for a single packing and is highly relevant for large and complicated meshes.

## Type of change
* New feature (non-breaking change which adds functionality)

Change summary:
---------------
* import imagewriter only if needed
* rename variables in `BuildGrid_trimesh`
* speed up grid creation using numpy and vectorization
* remove duplicate imports
* add description to progress bar
* create new function for vectorized point checking

Steps to Verify:
----------------
1.  `conda activate autopack`
2. `pack -r cellpack/tests/recipes/v2/test_mesh_image_gradient.json -c cellpack/tests/packing-configs/test_mesh_config.json`

Screenshots:
----------------
### Time on Main branch: 
image
image

### Time on updated branch:
image
image

### Surface points comparison (with a coarser grid for better visualization):
Main branch: 
image

Updated branch:
image

Keyfiles (delete if not relevant):
-----------------------
1. `cellpack/autopack/Compartment.py`

### CODE REVIEW

File: cellpack/autopack/Compartment.py [hunk 1]
```diff
@@ -1299,42 +1307,46 @@ def BuildGrid_trimesh(self, env, grdPos, vSurfaceArea, srfPts, idarray, mesh_sto
...
+        point_compartment_ids = compartment_ids[points_in_encap_sphere]
+        point_ids_to_assign = points_in_encap_sphere[point_compartment_ids == 0]
+        point_positions = master_grid_positions[point_ids_to_assign]
+
+        # check surface points
+        points_on_surface = trimesh_grid_surface.is_filled(point_positions)
+        compartment_ids[point_ids_to_assign[points_on_surface]] = number
+
+        # check inside points
+        points_in_mesh = mesh_store.contains_points_mesh(self.gname, point_positions)
```
Comment (rugeli) [inline cellpack/autopack/Compartment.py:1329]: The logic of the changes you made to check for inside points and set the compartment id is so clear!

File: cellpack/autopack/MeshStore.py [hunk 2]
```diff
@@ -264,6 +264,12 @@ def contains_points(self, geomname, points):
             return intersector.contains_points(points)
         return [False]
 
+    def contains_points_mesh(self, geomname, points):
+        mesh = self.get_object(geomname)
+        if mesh is not None:
+            return mesh.contains(points)
+        return numpy.full(len(points), False)
```
Comment (rugeli) [inline cellpack/autopack/MeshStore.py:271]: Good idea to create a boolean numpy array for points checking!

**Review by rugeli — APPROVED** [review]

File: cellpack/autopack/Analysis.py [hunk 3]
```diff
@@ -2354,6 +2353,8 @@ def pack_one_seed(
                 plt.close()  # closes the current figure
 
         if image_export_options is not None:
+            from cellpack.autopack.writers.ImageWriter import ImageWriter
+
```
Comment (meganrm) [inline cellpack/autopack/Analysis.py:2357]: why did you move this here?

File: cellpack/tests/packing-configs/test_mesh_config.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "place_method": "spheresSST",
+    "save_analyze_result": true,
+    "load_from_grid_file": false,
+    "show_progress_bar": true,
+    "show_grid_plot": true,
+    "spacing": null,
+    "parallel": false,
+    "use_periodicity": false,
+    "show_sphere_trees": true
+}
```
Comment (meganrm) [inline cellpack/tests/packing-configs/test_mesh_config.json:18]: how is this different than the other test configs we have?

Comment (mogres) [reply cellpack/tests/packing-configs/test_mesh_config.json:18]: This config re-creates the grid by default, and also shows the grid information in the simularium file. This is useful for debugging recipes involving meshes.

File: cellpack/autopack/Analysis.py (hunk 3 above)
Comment (mogres) [reply cellpack/autopack/Analysis.py:2357]: I ran a profiler on cellpack to identify time consuming steps and the import of the `ImageWriter` containing `aicsimageio` etc. took up a chunk of time even if images were not actually imported. It's not a very significant time saving so we could move it back to the top to be stylistically consistent.

Comment (meganrm) [reply cellpack/autopack/Analysis.py:2357]: I'm in favor of having imports at the top if it's not going to save very much time

**Review by meganrm — APPROVED** [review]

File: cellpack/autopack/MeshStore.py [hunk 5]
```diff
@@ -264,6 +266,22 @@ def contains_points(self, geomname, points):
             return intersector.contains_points(points)
         return [False]
```
Comment (meganrm) [inline cellpack/autopack/MeshStore.py:268]: Delete this function above since we're no longer using
